### PR TITLE
Use cached methods SentryDebugImageProvider of the Cocoa SDK

### PR DESCRIPTION
**Pull request by antonis**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Replaces `getDebugImagesForAddresses:isCrash:` with `getDebugImagesForImageAddressesFromCache` from https://github.com/getsentry/sentry-cocoa/pull/4435 to speed up getBinaryImages for finishing transactions and capturing events

## :bulb: Motivation and Context

Fixes https://github.com/getsentry/sentry-react-native/issues/4169

## :green_heart: How did you test it?
Manual tests, CI
Added tests in https://github.com/getsentry/sentry-cocoa/pull/4460

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: packages/core/ios/RNSentry.h [hunk 1]
```diff
@@ -14,7 +14,7 @@
 typedef int (*SymbolicateCallbackType)(const void *, Dl_info *);
 
 @interface SentryDebugImageProvider ()
-- (NSArray<SentryDebugMeta *> * _Nonnull)getDebugImagesForAddresses:(NSSet<NSString *> * _Nonnull)addresses isCrash:(BOOL)isCrash;
+- (NSArray<SentryDebugMeta *> * _Nonnull)getDebugImagesForImageAddressesFromCache:(NSSet<NSString *> * _Nonnull)addresses;
```
Comment (antonis) [inline packages/core/ios/RNSentry.h:17]: When [the updated header file(`SentryDebugImageProvider+HybridSDKs.h`)](https://github.com/getsentry/sentry-cocoa/pull/4460) is released in the next cocoa version we may remove this declaration completely.

**Comment by lucas-zimerman** [comment]
Does this PR depend on the new cocoa release before being merge? if so I would add an label `blocked` on it.
Otherwaise, the PR looks good to me!

**Comment by antonis** [comment]
Thank you for reviewing @lucas-zimerman 🙇 

> Does this PR depend on the new cocoa release before being merge? 

At the current state the PR does not depend on a future Cocoa release. I've [added a note though](https://github.com/getsentry/sentry-react-native/pull/4194#discussion_r1820459792) to remove  the private method declaration when [the new hybrid sdk header](https://github.com/getsentry/sentry-cocoa/pull/4460/files#diff-542f98f10ae9e90fc89c8a10149d4c6dd7da644437cdc1fbf472e4ed5d0ee613R32) get's released in Cocoa. I think this can be done in another PR when Cocoa 8.40 is out.

**Comment by antonis** [comment]
> At the current state the PR does not depend on a future Cocoa release. I've https://github.com/getsentry/sentry-react-native/pull/4194#discussion_r1820459792 to remove the private method declaration when [the new hybrid sdk header](https://github.com/getsentry/sentry-cocoa/pull/4460/files#diff-542f98f10ae9e90fc89c8a10149d4c6dd7da644437cdc1fbf472e4ed5d0ee613R32) get's released in Cocoa. I think this can be done in another PR when Cocoa 8.40 is out.

With [Cocoa 8.40.0 merged](https://github.com/getsentry/sentry-react-native/pull/4239) I removed the private function declaration (48fdebdb1a6f068fbcbcba026dbd43f4bdc60e94).
Ready for another pass.

**Review by krystofwoldrich — APPROVED** [review]
LGTM 🚀 Thank you

**Pull request merged**